server/src/main/java/org/elasticsearch/index/engine/EngineConfig.java

```diff
@@ -126,6 +127,11 @@ public EngineConfig(ShardId shardId, String allocationId, ThreadPool threadPool,
                         List<ReferenceManager.RefreshListener> internalRefreshListener, Sort indexSort,
                         CircuitBreakerService circuitBreakerService, LongSupplier globalCheckpointSupplier,
                         LongSupplier primaryTermSupplier, TombstoneDocSupplier tombstoneDocSupplier) {
+        if (indexSettings.getIndexVersionCreated().onOrAfter(Version.V_7_0_0)
```

Comment:
I think this is too late to be rejecting this setting. It means the index will already be created and in the cluster state, but all shards will fail during engine creation. The user will be left with a red index.

Reply:
Thanks. I moved to ```createIndexService```. Could you please review again?